File: Classes/WMISet/WMISet/Examples/cbuilder/WmiMethods/NetworkConfig/FrmStaticAddressU.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "FrmStaticAddressU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TFrmStaticAddress *FrmStaticAddress;
//---------------------------------------------------------------------------
__fastcall TFrmStaticAddress::TFrmStaticAddress(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TFrmStaticAddress::edtIPAddressChange(TObject *Sender)
{
  SetButtonState();
}

//---------------------------------------------------------------------------
void __fastcall TFrmStaticAddress::SetButtonState() {
  btnSetStaticAddress->Enabled = (edtIPAddress->Text.Trim() != "") &&
                                 (edtDefaultGateway->Text.Trim() != "") &&
                                 (edtIPSubnetMask->Text.Trim() != "");
};

File: Classes/WMISet/WMISet/Examples/delphi/WmiMethods/NetworkConfig/FrmStaticAddressU.pas
unit FrmStaticAddressU;

interface

uses
  Windows, Messages, SysUtils,
  {$IFDEF Delphi6}
  Variants,
  {$ENDIF}
  Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls;

type
  TFrmStaticAddress = class(TForm)
    lblIPAddress: TLabel;
    edtIPAddress: TEdit;
    lblSubnetMask: TLabel;
    edtIPSubnetMask: TEdit;
    btnSetStaticAddress: TButton;
    btnCancel: TButton;
    Label1: TLabel;
    edtDefaultGateway: TEdit;
    procedure edtIPAddressChange(Sender: TObject);
  private
    procedure SetButtonState;
    { Private declarations }
  public
    { Public declarations }
  end;

var
  FrmStaticAddress: TFrmStaticAddress;

implementation

{$R *.dfm}

procedure TFrmStaticAddress.edtIPAddressChange(Sender: TObject);
begin
  SetButtonState;
end;

procedure TFrmStaticAddress.SetButtonState;
begin
  btnSetStaticAddress.Enabled := (Trim(edtIPAddress.text) <> '') and
                                 (Trim(edtIPSubnetMask.text) <> '') and
                                 (Trim(edtDefaultGateway.Text) <> '');
end;

end.
